internal/common/validation/job.go

```diff
@@ -3,9 +3,77 @@ package validation
 import (
 	"fmt"
 
+	"github.com/pkg/errors"
+
+	"github.com/G-Research/armada/internal/common/armadaerrors"
+
 	"github.com/G-Research/armada/pkg/api"
 )
 
+func ValidateJobRequestItemPodSpec(r *api.JobSubmitRequestItem) error {
```

Comment:
Is there a reason why we didn't add tests for these functions?

Reply:
I'll add tests